README.md

```diff
@@ -1,101 +1,94 @@
-  bundle:
-    name: resolved-654adh
-    spec:
-      class: rukpack.io/k8s
-      refs:
-      - file://content
-      volumeMounts:
-      - mountPath: /content
-        configMap:
-          name: resolved-654adh-content
-          namespace: olm
+  image: my-bundle@sha256:xyz123
+  provisionerClassName: core.rukpak.io/plain-v0
 ```
 
-### ProvisionerClass
-`ProvisionerClass` defines a configuration for a `Provisioner`. Provisioners are controllers that understand `Instance`, `Bundle`, and `ProvisionerClass` APIs and take action.
+### BundleInstance
 
-Each `Provisioner` has a unique id. For example, a `Provisioner` that understands bundles composed of arbitrary Kubernetes YAML manifests can have the id: rukpack.io/k8s.
+The `BundleInstance` API points to a Bundle and indicates that it should be “active”. This includes pivoting from older
+versions of an active bundle.`BundleInstance` may also include an embedded spec for a desired Bundle.
 
-A `ProvisionerClass` specifies a specific configuration of provisioning (i.e. settings to interpret Instance and Bundle APIs). 
-A `Provisioner` will only operate on Instance and Bundle objects that reference a `ProvisionerClass` that contain the provisioner’s unique id.
+Much like Pods stamp out instances of container images, `BundleInstances` stamp out an instance of
+Bundles. `BundleInstance` can be seen as a generalization of the Pod concept.
 
-If this seems familiar, it is the same pattern that is used by StorageClass / PeristentVolume / PersistentVolumeClaim in Kubernetes. This design is meant to be extendable 
-to support `Bundles` and `Instances` backed by different content sources. 
+The specifics of how an `BundleInstance` makes changes to a cluster based on a referenced `Bundle` is defined by the
+`Provisioner` that is configured to watch that `BundleInstance`.
+
+Example BundleInstance configured to work with the [plain-v0 provisioner](provisioner/plain-v0/README.md).
 
-Example ProvisionerClass
 ```yaml
 apiVersion: core.rukpak.io/v1alpha1
-kind: ProvisionerClass
+kind: BundleInstance
 metadata:
-  name: <good name>
-provisioner: <unique id>
-# parameters has no schema, is provisioner-specific
-parameters: {}
+  name: my-bundle-instance
+spec:
+  bundleName: my-bundle
+  provisionerClassName: core.rukpak.io/plain-v0
 ```
 
-## Contributing 
+### Provisioner
+
+A Provisioner is a controller that understands `BundleInstance` and `Bundle` APIs and can take action.
+Each `Provisioner` is assigned a unique ID, and is responsible for reconciling a `Bundle` and `BundleInstance` with
+a `spec.provisionerClassName` that matches that particular ID.
 
-RukPak is a community-based open source project, and all are welcome to get involved. 
\ No newline at end of file
+For example, in this repository the [plain-v0](provisioner/plain-v0/README.md) provisioner is implemented.
+The `plain-v0` provisioner is able to unpack a given `plain+v0` bundle onto a cluster and then instantiate it, making
```

Comment:
non-blocking: `plain-v0` followed by `plain+v0` may be confusing, but we've been having that conversation for a bit.

Reply:
Yes, I agree 😄 , this is the current state of our naming conventions though. When the `plain+v0` bundle spec becomes available, and this sentence can link to that document, hopefully it will be less confusing.